profiles/rawls-service-perimeters/terraform/rawls-service-perimeters/variables.tf.ctmpl

```diff
@@ -17,6 +17,21 @@ variable "access_policy_name" {
   default = "{{env "ACCESS_POLICY_NAME"}}"
 }
 
+# There may be a better way of determining the Terraform SA, but this feels
+# slightly better than encoding the value twice.
+{{ with $creds_json := file "/default.sa.json" | parseJSON }}
+
+{{ $terraform_sa := "FATAL: must use service-account credentials" }}
+{{ if eq "service_account" (index $creds_json "type") }}
+  {{ $terraform_sa = index $creds_json "client_email" }}
+{{ end }}
+
+variable "terraform_sa" {
+  type = string
+  default = "{{ $terraform_sa }}"
+}
+{{ end }}
+
```

Comment:
I think that's the right Idea, but the variable definition needs to be done at the Terraform level, not in consul-template, as the consul-template step happens before the service account JSON exists:

```
locals {
  terraform_sa = jsondecode(file("default.sa.json")).client_email
}
```

This is the plan when I make that change:
```
Terraform will perform the following actions:

  # google_bigquery_dataset.bigquery-sink-dataset["terra_prod_aou_preprod"] will be updated in-place
  ~ resource "google_bigquery_dataset" "bigquery-sink-dataset" {
        creation_time                   = 1582924474963
        dataset_id                      = "WorkspaceBigQueryLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "BigQuery audit log sink for projects in Terra perimeter terra_prod_aou_preprod"
        etag                            = "Y3jq4O+bvEyu8TVHiPqSNg=="
        id                              = "projects/fc-aou-logs-preprod/datasets/WorkspaceBigQueryLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_preprod"
        }
        last_modified_time              = 1582924474963
        location                        = "US"
        project                         = "fc-aou-logs-preprod"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-preprod/datasets/WorkspaceBigQueryLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f1030649683602-299296@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f1030649683602-299296@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.bigquery-sink-dataset["terra_prod_aou_prod"] will be updated in-place
  ~ resource "google_bigquery_dataset" "bigquery-sink-dataset" {
        creation_time                   = 1582924474361
        dataset_id                      = "WorkspaceBigQueryLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "BigQuery audit log sink for projects in Terra perimeter terra_prod_aou_prod"
        etag                            = "pDJ37DzLabYBt8R61rMU8A=="
        id                              = "projects/fc-aou-logs-prod/datasets/WorkspaceBigQueryLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_prod"
        }
        last_modified_time              = 1582924474361
        location                        = "US"
        project                         = "fc-aou-logs-prod"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-prod/datasets/WorkspaceBigQueryLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f300547813290-440415@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f300547813290-440415@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.bigquery-sink-dataset["terra_prod_aou_stable"] will be updated in-place
  ~ resource "google_bigquery_dataset" "bigquery-sink-dataset" {
        creation_time                   = 1582924474464
        dataset_id                      = "WorkspaceBigQueryLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "BigQuery audit log sink for projects in Terra perimeter terra_prod_aou_stable"
        etag                            = "Y2+ITsLZp4OT+IfU8kPiGg=="
        id                              = "projects/fc-aou-logs-stable/datasets/WorkspaceBigQueryLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_stable"
        }
        last_modified_time              = 1582924474464
        location                        = "US"
        project                         = "fc-aou-logs-stable"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-stable/datasets/WorkspaceBigQueryLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f96867205717-390018@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f96867205717-390018@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.bigquery-sink-dataset["terra_prod_aou_staging"] will be updated in-place
  ~ resource "google_bigquery_dataset" "bigquery-sink-dataset" {
        creation_time                   = 1582924474461
        dataset_id                      = "WorkspaceBigQueryLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "BigQuery audit log sink for projects in Terra perimeter terra_prod_aou_staging"
        etag                            = "FtyLvsrlu3EQ3Pu2pKCElQ=="
        id                              = "projects/fc-aou-logs-staging/datasets/WorkspaceBigQueryLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_staging"
        }
        last_modified_time              = 1582924474461
        location                        = "US"
        project                         = "fc-aou-logs-staging"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-staging/datasets/WorkspaceBigQueryLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f727685043294-517572@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f727685043294-517572@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.dataproc-sink-dataset["terra_prod_aou_preprod"] will be updated in-place
  ~ resource "google_bigquery_dataset" "dataproc-sink-dataset" {
        creation_time                   = 1582924474555
        dataset_id                      = "WorkspaceDataprocLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "Dataproc audit log sink for projects in Terra perimeter terra_prod_aou_preprod"
        etag                            = "VJ2wUFxQvKVcFRT75jWiSQ=="
        id                              = "projects/fc-aou-logs-preprod/datasets/WorkspaceDataprocLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_preprod"
        }
        last_modified_time              = 1582924474555
        location                        = "US"
        project                         = "fc-aou-logs-preprod"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-preprod/datasets/WorkspaceDataprocLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f1030649683602-299296@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f1030649683602-299296@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.dataproc-sink-dataset["terra_prod_aou_prod"] will be updated in-place
  ~ resource "google_bigquery_dataset" "dataproc-sink-dataset" {
        creation_time                   = 1582924474458
        dataset_id                      = "WorkspaceDataprocLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "Dataproc audit log sink for projects in Terra perimeter terra_prod_aou_prod"
        etag                            = "w4yFJFxAkLzn4IQfs8OWlQ=="
        id                              = "projects/fc-aou-logs-prod/datasets/WorkspaceDataprocLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_prod"
        }
        last_modified_time              = 1582924474458
        location                        = "US"
        project                         = "fc-aou-logs-prod"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-prod/datasets/WorkspaceDataprocLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f300547813290-440415@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f300547813290-440415@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.dataproc-sink-dataset["terra_prod_aou_stable"] will be updated in-place
  ~ resource "google_bigquery_dataset" "dataproc-sink-dataset" {
        creation_time                   = 1582924474461
        dataset_id                      = "WorkspaceDataprocLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "Dataproc audit log sink for projects in Terra perimeter terra_prod_aou_stable"
        etag                            = "yF7VlZKgju8H1vheISS5zw=="
        id                              = "projects/fc-aou-logs-stable/datasets/WorkspaceDataprocLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_stable"
        }
        last_modified_time              = 1582924474461
        location                        = "US"
        project                         = "fc-aou-logs-stable"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-stable/datasets/WorkspaceDataprocLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f96867205717-390018@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f96867205717-390018@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.dataproc-sink-dataset["terra_prod_aou_staging"] will be updated in-place
  ~ resource "google_bigquery_dataset" "dataproc-sink-dataset" {
        creation_time                   = 1582924475166
        dataset_id                      = "WorkspaceDataprocLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "Dataproc audit log sink for projects in Terra perimeter terra_prod_aou_staging"
        etag                            = "f0pNzXsXesAc3tv2cfQFfQ=="
        id                              = "projects/fc-aou-logs-staging/datasets/WorkspaceDataprocLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_staging"
        }
        last_modified_time              = 1582924475166
        location                        = "US"
        project                         = "fc-aou-logs-staging"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-staging/datasets/WorkspaceDataprocLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f727685043294-517572@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f727685043294-517572@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.storage-sink-dataset["terra_prod_aou_preprod"] will be updated in-place
  ~ resource "google_bigquery_dataset" "storage-sink-dataset" {
        creation_time                   = 1582924474462
        dataset_id                      = "WorkspaceStorageLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "Storage audit log sink for projects in Terra perimeter terra_prod_aou_preprod"
        etag                            = "BxWDEl4wS43dB0D1xWyL+Q=="
        id                              = "projects/fc-aou-logs-preprod/datasets/WorkspaceStorageLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_preprod"
        }
        last_modified_time              = 1582924474462
        location                        = "US"
        project                         = "fc-aou-logs-preprod"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-preprod/datasets/WorkspaceStorageLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f1030649683602-299296@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f1030649683602-299296@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.storage-sink-dataset["terra_prod_aou_prod"] will be updated in-place
  ~ resource "google_bigquery_dataset" "storage-sink-dataset" {
        creation_time                   = 1582924474461
        dataset_id                      = "WorkspaceStorageLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "Storage audit log sink for projects in Terra perimeter terra_prod_aou_prod"
        etag                            = "MHrjwmBJVj0uPngKewf6vQ=="
        id                              = "projects/fc-aou-logs-prod/datasets/WorkspaceStorageLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_prod"
        }
        last_modified_time              = 1582924474461
        location                        = "US"
        project                         = "fc-aou-logs-prod"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-prod/datasets/WorkspaceStorageLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f300547813290-440415@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f300547813290-440415@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.storage-sink-dataset["terra_prod_aou_stable"] will be updated in-place
  ~ resource "google_bigquery_dataset" "storage-sink-dataset" {
        creation_time                   = 1582924475060
        dataset_id                      = "WorkspaceStorageLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "Storage audit log sink for projects in Terra perimeter terra_prod_aou_stable"
        etag                            = "qz+xnQOlvzNsM+e5UTSS5g=="
        id                              = "projects/fc-aou-logs-stable/datasets/WorkspaceStorageLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_stable"
        }
        last_modified_time              = 1582924475060
        location                        = "US"
        project                         = "fc-aou-logs-stable"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-stable/datasets/WorkspaceStorageLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f96867205717-390018@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f96867205717-390018@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

  # google_bigquery_dataset.storage-sink-dataset["terra_prod_aou_staging"] will be updated in-place
  ~ resource "google_bigquery_dataset" "storage-sink-dataset" {
        creation_time                   = 1582924475163
        dataset_id                      = "WorkspaceStorageLogs"
        default_partition_expiration_ms = 0
        default_table_expiration_ms     = 0
        delete_contents_on_destroy      = false
        description                     = "Storage audit log sink for projects in Terra perimeter terra_prod_aou_staging"
        etag                            = "ALtF9crKoL3RVPM98F0AXQ=="
        id                              = "projects/fc-aou-logs-staging/datasets/WorkspaceStorageLogs"
        labels                          = {
            "perimeter" = "terra_prod_aou_staging"
        }
        last_modified_time              = 1582924475163
        location                        = "US"
        project                         = "fc-aou-logs-staging"
        self_link                       = "https://www.googleapis.com/bigquery/v2/projects/fc-aou-logs-staging/datasets/WorkspaceStorageLogs"

      + access {
          + role          = "EDITOR"
          + user_by_email = "f727685043294-517572@gcp-sa-logging.iam.gserviceaccount.com"
        }
      - access {
          - role          = "OWNER" -> null
          - user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com" -> null
        }
      - access {
          - role          = "WRITER" -> null
          - user_by_email = "f727685043294-517572@gcp-sa-logging.iam.gserviceaccount.com" -> null
        }
      + access {
          + role          = "OWNER"
          + user_by_email = "terra-prod-terraform-sa@firecloud-org-terraform-base.iam.gserviceaccount.com"
        }
    }

Plan: 0 to add, 12 to change, 0 to destroy.
```

Reply:
Thanks, sounds good. We should see if I can get dev vault access now (since I have a Broad account) - that would help me actually validate these changes e2e.

Please push this change to my branch when you're able - and rerun with my latest commits. Our goal here is to see 0 diff in the TF plan.